environment/integration/helpers.go

```diff
@@ -169,7 +203,7 @@ func initializeDaggerOnce(t *testing.T) {
 	})
 
 	if daggerErr != nil {
-		t.Skipf("Skipping test - Dagger not available: %v", daggerErr)
+		t.Fatalf("Dagger not available: %v", daggerErr)
```

Comment:
🐛 **Bug (Medium):** This change from t.Skipf to t.Fatalf will cause tests to fail hard instead of gracefully skipping when Docker/Dagger isn't available. This could break CI environments where Docker might not be available, whereas the previous behavior allowed tests to skip gracefully.

```suggestion
		t.Skipf("Skipping test - Dagger not available: %v", daggerErr)
```